### ISSUE TITLE
»arguments« configuration setting is broken

### ISSUE THREAD

**Issue by quchen** [ndmitchell/hlint#340]
Having the `arguments` flag present with non-empty content in `.hlint.yaml` produces an error,
```
echo "main = id (pure ())" > Main.hs
echo "- arguments: [--color]" > .hlint.yaml
hlint .
```
```
hlint: .: openFile: inappropriate type (is a directory)
```

The issue seems independent of the flag in the list; `foo` instead of `--color` does not change the behaviour. Removing `.hlint.yaml` or emptying its contents produces the expected result,

```
./Main.hs:1:8: Warning: Evaluate
Found:
  id (pure ())
Why not:
  (pure ())

1 hint
```

**Comment by ndmitchell** [comment]
Thanks for the report - interesting indeed. I'll investigate.

**Comment by sergv** [comment]
It seems to me that the issue is caused by the following line: https://github.com/ndmitchell/hlint/blob/master/src/HLint.hs#L150. If yaml file contains `arguments` declaration, the `readAllSettings` function will try to parse command with extended list of arguments and return new command instead of old one that was specified to `runHints`. The issue is that old command has files already resolved, while newly parsed command has no such thing.

Consider my local example:

The old command than gets into `runHints`
```haskell
CmdMain
  { cmdFiles =
      [ "./Setup.hs"
      , "./doc-svg-gen/Main.hs"
      , "./doc-svg-gen/Setup.hs"
      , "./src/Language/Haskell/Brittany.hs"
      , "./src/Language/Haskell/Brittany/Backend.hs"
      , "./src/Language/Haskell/Brittany/BackendUtils.hs"
      , "./src/Language/Haskell/Brittany/Config.hs"
      , "./src/Language/Haskell/Brittany/ExactPrintUtils.hs"
      , "./src/Language/Haskell/Brittany/LayouterBasics.hs"
      , "./src/Language/Haskell/Brittany/Prelude.hs"
      , "./src/Language/Haskell/Brittany/Types.hs"
      , "./src/Language/Haskell/Brittany/Utils.hs"
      , "./src/Language/Haskell/Brittany/Config/Types.hs"
      , "./src/Language/Haskell/Brittany/Layouters/Decl.hs"
      , "./src/Language/Haskell/Brittany/Layouters/Expr.hs"
      , "./src/Language/Haskell/Brittany/Layouters/Pattern.hs"
      , "./src/Language/Haskell/Brittany/Layouters/Stmt.hs"
      , "./src/Language/Haskell/Brittany/Layouters/Type.hs"
      , "./src/Language/Haskell/Brittany/Transformations/Alt.hs"
      , "./src/Language/Haskell/Brittany/Transformations/Columns.hs"
      , "./src/Language/Haskell/Brittany/Transformations/Floating.hs"
      , "./src/Language/Haskell/Brittany/Transformations/Indent.hs"
      , "./src/Language/Haskell/Brittany/Transformations/Par.hs"
      , "./src-brittany/Main.hs"
      , "./src-idemtests/cases/LayoutBasics.hs"
      , "./src-literatetests/Main.hs"
      , "./src-unittests/AsymptoticPerfTests.hs"
      , "./src-unittests/TestMain.hs"
      , "./src-unittests/TestUtils.hs"
      ]
  , cmdReports = []
  , cmdGivenHints = []
  , cmdWithHints = []
  , cmdColor = Auto
  , cmdThreads = 1
  , cmdIgnore = []
  , cmdShowAll = False
  , cmdExtension = [ "hs" , "lhs" ]
  , cmdLanguage = []
  , cmdCross = False
  , cmdFindHints = []
  , cmdDataDir =
      "/home/sergey/projects/haskell/projects/dev-tools/hlint/.stack-work/install/x86_64-linux/lts-8.11/8.0.2/share/x86_64-linux-ghc-8.0.2/hlint-2.0.6"
  , cmdDefault = False
  , cmdPath = [ "." ]
  , cmdCppDefine = []
  , cmdCppInclude = []
  , cmdCppFile = []
  , cmdCppSimple = False
  , cmdCppAnsi = False
  , cmdJson = False
  , cmdNoSummary = False
  , cmdOnly = []
  , cmdNoExitCode = False
  , cmdSerialise = False
  , cmdRefactor = False
  , cmdRefactorOptions = ""
  , cmdWithRefactor = ""
  }
```

Newly parsed command after line 150:
```haskell
CmdMain
  { cmdFiles = [ "." ]
  , cmdReports = []
  , cmdGivenHints = []
  , cmdWithHints = []
  , cmdColor = Auto
  , cmdThreads = 1
  , cmdIgnore = []
  , cmdShowAll = False
  , cmdExtension = [ "hs" , "lhs" ]
  , cmdLanguage = []
  , cmdCross = False
  , cmdFindHints = []
  , cmdDataDir =
      "/home/sergey/projects/haskell/projects/dev-tools/hlint/.stack-work/install/x86_64-linux/lts-8.11/8.0.2/share/x86_64-linux-ghc-8.0.2/hlint-2.0.6"
  , cmdDefault = False
  , cmdPath = [ "." ]
  , cmdCppDefine = []
  , cmdCppInclude = [ "srcinc" ]
  , cmdCppFile = []
  , cmdCppSimple = False
  , cmdCppAnsi = False
  , cmdJson = False
  , cmdNoSummary = False
  , cmdOnly = []
  , cmdNoExitCode = False
  , cmdSerialise = False
  , cmdRefactor = False
  , cmdRefactorOptions = ""
  , cmdWithRefactor = ""
  }
```

**Comment by ndmitchell** [comment]
Fixed in hlint-2.0.7.

**Comment by quchen** [comment]
Thanks! :-)
